vtysh/vtysh.c

```diff
@@ -2396,14 +2397,18 @@ static int vtysh_exit(struct vty *vty)
 		vty->node = cnode->parent_node;
 
 	if (vty->node == CONFIG_NODE) {
+		bool locked = vty->vtysh_file_locked;
```

Comment:
What's the point of adding a bool here?  Why isn't the `if (vty->vtysh_file_locked)` sufficient?

Reply:
`vty->vtysh_file_locked` will be overwritten by `vtysh_execute("end")` on the next line. We need to remember it.